src/config/config.ts

```diff
@@ -135,4 +144,44 @@ export function initializeAPIClients(){
 
 }
 
+export function initializeConfiguration(){
+    // @ts-ignore: ENV_* are defined in webpack.config.js
+
+    // handle localStorage
+    // LOCAL STORAGE TRUMPS EVERYTHING EXCEPT WHAT'S ORIGINALLY SET IN JSP
+    if (localStorage.frontendConfig) {
+        try {
+            updateConfig(JSON.parse(localStorage.frontendConfig));
+            console.log("Using localStorage.frontendConfig (overriding window.frontendConfig): " + localStorage.frontendConfig);
+        } catch (err) {
+            // ignore
+            console.log("Error parsing localStorage.frontendConfig")
+        }
+    }
+
+    // @ts-ignore: ENV_* are defined in webpack.config.js
+    const APIROOT = `//${trimTrailingSlash(ENV_CBIOPORTAL_URL)}/`;
 
+    // we want to respect frontUrl if it is already set (case where localdist is true)
+    // @ts-ignore: ENV_* are defined in webpack.config.js
+    const frontendUrl = config.frontendUrl || (/\/\/localhost:3000/.test(win.location.href)) ? "//localhost:3000/" : `//${ENV_CBIOPORTAL_URL}/`;
+
+    const configServiceUrl = config.configurationServiceUrl || APIROOT;
+
+    const envConfig: Partial<IAppConfig> = {
+        apiRoot:APIROOT,
+        frontendUrl:frontendUrl,
+        configurationServiceUrl:APIROOT,
```

Comment:
should that `configurationServiceUrl:APIROOT` be `configurationServiceUrl: configServiceUrl `?